components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/mediasession/GeckoMediaSession.kt

```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.mediasession
+
+import mozilla.components.concept.engine.mediasession.MediaSession
+import org.mozilla.geckoview.MediaSession as GeckoViewMediaSession
+
+internal class GeckoMediaSession(
+    private val geckoViewMediaSession: GeckoViewMediaSession
+) : MediaSession() {
```

Comment:
I wonder if we even need this `MediaSession` interface and the `GeckoMediaSession` implementations: In the reducer we seem to only take things like the `controller` or the `isActive` flag, so maybe we could just pass them through the observer to the store and keep all the media session details in GV?

Reply:
Updated to remove `GeckoMediaSession`.  Kept `MediaSession` to keep controller interface and data classes in one place.  What do you think?